datumaro/plugins/coco_format/extractor.py

```diff
@@ -215,47 +205,54 @@ def _load_pan_mask(path):
         mask = bgr2index(mask)
         return mask
 
+    @define(on_setattr=False)
```

Comment:
What does `on_setattr=False` do? According to the reference, it should be a callable.

Reply:
And judging by the usage of the class, `attrs.frozen` might be a better fit anyway.

Reply:
Strangely, I remember I've read it somewhere, but docs are silent about it. Maybe, we don't need this at all in this class.

It disables `setattr` checks for new-style declaration (via `define`):

``` python
def myconv(v):
    print('conv')
    return v

def myvalid(a, b, v):
    print('valid')

@define(on_setattr=False)
class A:
    x = field(default=5, converter=myconv, validator=myvalid)

    def __init__(self):
        self.x = 10

    def f(self):
        print(self.x)

@define(on_setattr=None)
class B:
    x = field(default=5, converter=myconv, validator=myvalid)

    def __init__(self):
        self.x = 10

    def f(self):
        print(self.x)

@attrs(init=False)
class C:
    x = field(default=5, converter=myconv, validator=myvalid)

    def __init__(self):
        self.x = 10

    def f(self):
        print(self.x)

print('A1')
print(A())

print('A2')
A().x = 7

print('B1')
print(B())

print('B2')
B().x = 7

print('C1')
print(C())

print('C2')
C().x = 7

>>>
A1
A(x=10)
A2

B1
conv
valid
B(x=10)
B2

conv
valid
conv
valid
C1
C(x=10)
C2
```

Reply:
Removed

Reply:
FWIW, I think the documented way of achieving this effect is `on_setattr=attrs.setters.NO_OP`, although I don't think it's needed here.